wolfssl/internal.h

```diff
@@ -4487,6 +4491,9 @@ struct WOLFSSL {
     byte clientFinished[TLS_FINISHED_SZ];
     byte serverFinished[TLS_FINISHED_SZ];
 #endif
+#if defined(WOLFSSL_IOTSAFE) && defined(HAVE_PK_CALLBACKS)
+    IOTSAFE iotsafe;
```

Comment:
Do we need to have the `IOTSAFE` struct in the WOLFSSL object? Can't we use the custom IO callback context and keep this in the application? See `wolfSSL_SetIOReadCtx` and `wolfSSL_SetIOWriteCtx`?

Reply:
I think we should consider, in this case, to keep the additional fields there. The IOTSAFE object only contains per-session information about IoT-Safe key slots. 

It makes sense to have this object built-in, because of the lifetime of the object itself, i.e. if an application should handle the IOTSAFE object externally, the object will have to be allocated, associated to the SSL object (via PK callbacks) and then freed upon `wolfSSL_free()`.

This choice simplifies the iotsafe API associating to ctx/ssl objects, and keep the information hidden and accessible only when iotsafe is enabled on the session, via `wolfSSL_iotsafe_on()`.